### PR TITLE
CorpusViewer: Show tokens & tags

### PR DESCRIPTION
**Pull request by nikicc**
Introduced changes to CorpusViewer:
- Allow showing of tokens and tags. To make fancy styling possible `QTextEdit` was replaced with `gui.WebView`. `mark.js` was also introduced since QTextEdit does not allow for highlighting by regular expressions. Also `mark.js` allows for highlighting only matches in `search_features` and disabling highlighting on feature names 🎉 🎆 . 
- Enhance `Info` box with info on preprocessing, number of tokens and types, POS tagging and ngrams.
- Allow reordering of Display features.
- Use new style warnings.

### CODE REVIEW

**Comment by ajdapretnar** [comment]
I like this. Tokens, however, should probably be written in the same style as features, that is lowercase and bold. Do we plan to include tokens in Display features or not? Also I presume POS tagging will be added?

**Comment by nikicc** [comment]
> I like this. Tokens, however, should probably be written in the same style as features, that is lowercase and bold. 

Sure. This is still WIP.

>Do we plan to include tokens in Display features or not? 

Not sure whether to put them in display features or somewhere else? Any suggestions?

> Also I presume POS tagging will be added?

Of course.

**Comment by ajdapretnar** [comment]
> Not sure whether to put them in display features or somewhere else?

Heh, it's a tricky one for all the reasons we discussed before, but I think it would indeed make the most sense to put it in Display features since it's what it is.

**Comment by nikicc** [comment]
> Heh, it's a tricky one for all the reasons we discussed before, but I think it would indeed make the most sense to put it in Display features since it's what it is.

Yes, you're probably right. Though we should be careful about POS tags. If we have one _feature_ for tokens and an other one for tags user could show just POS tags without tokens which does not makes much sense.

Will first work on the display part and meanwhile we can decide on where to put it 😄

**Comment by ajdapretnar** [comment]
Could we overlay the tokens with POS tags somehow? And kind of make POS tags not a display feature when Tokens are disabled as well? #thinkingoutloud :smile:

**Comment by nikicc** [comment]
> Could we overlay the tokens with POS tags somehow?

That's the plan.

> And kind of make POS tags not a display feature when Tokens are disabled as well?

I'm not sue if I like those kind of dependencies among features but then again, currently, I do not have a better suggestion 🙈

**Comment by nikicc** [comment]
@ajdapretnar regarding the error issues we discussed yesterday I think that for now we can leave this as it is. This only gets annoying if you use drag & drop for reordering. It is still an improvement over the current state where no reordering is possible — though not the optimal one — but it does not wreck anything for users that do not use it.

IMO this is ready for review. Perhaps @kernc can you also take a look?

File: orangecontrib/text/widgets/owcorpusviewer.py [hunk 1]
```diff
@@ -43,44 +32,53 @@ class OWCorpusViewer(OWWidget):
...
+        self.search_features = []       # two copies are needed since Display allows drag & drop
+        self.display_features = []
 
-        # ---- CONTROL AREA ----
-        # Filtering results.
+        # Info
         filter_result_box = gui.widgetBox(self.controlArea, 'Info')
-        self.info_all = gui.label(filter_result_box, self, 'All documents:')
-        self.info_fil = gui.label(filter_result_box, self, 'After filtering:')
+        self.info_docs = gui.label(filter_result_box, self, 'Documents:')
```
Comment (kernc) [inline orangecontrib/text/widgets/owcorpusviewer.py]: The way `gui.label()` can be used is:
```py
gui.label(filter_results_box, self, 'Documents: %(n_documents)d')
...
self.n_documents = 233  # the label updates automagically
```

**Pull request merged**